site/src/contexts/ProxyContext.tsx

```diff
@@ -6,25 +6,60 @@ import {
   createContext,
   FC,
   PropsWithChildren,
+  useCallback,
   useContext,
+  useEffect,
   useState,
 } from "react"
 import { ProxyLatencyReport, useProxyLatency } from "./useProxyLatency"
 
-interface ProxyContextValue {
+export interface ProxyContextValue {
+  // proxy is **always** the workspace proxy that should be used.
```

Comment:
because it has the lowest latency?

Reply:
Because it's the best choice given the current information.

If the user selection is "unhealthy" for example, then `proxy` will defer to being auto selected. The auto selection is based on latency yes.